src/data_sources.js

```diff
@@ -2125,6 +2127,7 @@ class data_source_bookmarks_merged extends data_source_bookmarks_base
         {
             // We're past the end.
             console.log("page", page, "beyond", max_page, rest);
+            this.bookmark_illust_ids[is_private][page] = [];
```

Comment:
I see what was happening.  When it loaded a page with no results it would append [undefined] and confuse things later.  This didn't happen to me because I have tons of both public and private bookmarks, so I rarely scroll through to the end of either of them.  Instead of adding an empty entry, I think I'll check for this in continue_loading_page_internal instead, so it isn't confused by a page not being there.

Reply:
Oh, I thought it is browser dependent, like other browsers can concatenate null as empty list